app/src/main/java/de/westnordost/streetcomplete/quests/address/AddAddressStreetForm.kt

```diff
@@ -0,0 +1,179 @@
+                defaultName
+        )
+    }
+
+    private fun getRoadNameSuggestions(): List<MutableMap<String, String>> {
+        return roadNameSuggestionsDao.getNames(
+                geometryToMajorPoints(elementGeometry),
+                AddAddressStreet.MAX_DIST_FOR_ROAD_NAME_SUGGESTION
+        )
+    }
+
+    private fun geometryToMajorPoints(geometry: ElementGeometry): List<LatLon> {
+        return when(geometry) {
+            is ElementPolylinesGeometry -> {
+                val polyline = geometry.polylines.first()
+                listOf(polyline.first(), polyline.last())
+            }
+            is ElementPolygonsGeometry -> {
+                // return center and one of nodes from the way constructing area
+                listOf(geometry.center, geometry.polygons.first().last())
+            }
+            is ElementPointGeometry -> {
+                listOf(geometry.center)
+            }
+        }
+
+
+    }
+
+    override fun isRejectingClose() : Boolean {
+        // if the form is complete, we will reject close, unless it is still has only the default name
+        if (isFormComplete()) {
+            if (adapter.localizedNames.size == 1 && adapter.localizedNames.first().name == defaultName) {
+                return false
+            } else {
+                return true
+            }
+        } else {
+            return false
+        }
+    }
+
+    private fun switchToPlaceName() {
+        AlertDialog.Builder(activity!!)
+                .setTitle(R.string.quest_address_street_noStreet_confirmation_title)
+                .setPositiveButton(R.string.quest_address_street_noStreet_confirmation_positive) {_, _ -> switchToPlaceNameLayout() } //
+                .setNegativeButton(R.string.quest_generic_confirmation_no, null)
+                .show()
+    }
+
+    private fun switchToPlaceNameLayout() {
+        isPlaceName = true
+        setLayout(R.layout.quest_streetname_place)
+    }
+
+    private fun setLayout(layoutResourceId: Int) {
+        val view = setContentView(layoutResourceId)
+        val buttonNameSuggestions : View = view.findViewById(R.id.nameSuggestionsButton)
+        textField = view.findViewById(R.id.name)
+
+        //TODO - use actual place names
+        val nameSuggestionsMap = getRoadNameSuggestions()
+        val nameSuggestionsList = mutableListOf<String>()
+        for (NameSuggestion in nameSuggestionsMap) {
+            val name = NameSuggestion[""] ?: continue // just default language names
+            nameSuggestionsList += name
+        }
+
+        buttonNameSuggestions.setOnClickListener { v ->
+            showNameSuggestionsMenu(v, nameSuggestionsList) { selected -> ;
+                textField!!.setText(selected)
+            }
+        }
+        textField!!.addTextChangedListener(TextChangedWatcher { checkIsFormComplete() })
+    }
+
+    companion object {
+        private const val NAMES_DATA = "names_data"
+    }
```

Comment:
the companion object should be the very last thing in a class definition

Reply:
Fixed in https://github.com/westnordost/StreetComplete/pull/1782/commits/d7f9933ddf9577973fb30a3665ef2e220999b692